docs/current_docs/manuals/developer/documentation.mdx

```diff
@@ -0,0 +1,136 @@
+---
+slug: /manuals/developer/documentation
+displayed_sidebar: "current"
+toc_max_heading_level: 2
+title: "Inline Documentation"
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Inline Documentation
+
+Dagger modules and Dagger Functions should be documented so that descriptions are shown in the API and the CLI - for example, when calling `dagger functions` and `dagger call ... --help`.
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+The following code snippet shows how to add documentation for:
+
+- The whole module
+- Function methods
+- Function arguments
+
+```go file=./snippets/documentation/go/main.go
+```
+
+</TabItem>
+<TabItem value="Python">
+
+The following code snippet shows how to use Python's [documentation string conventions](https://peps.python.org/pep-0008/#documentation-strings) for adding descriptions to:
+- The whole [module](https://docs.python.org/3/tutorial/modules.html) or [package](https://docs.python.org/3/tutorial/modules.html#packages)
+- Object type classes (group of functions)
+- Function methods
+- Function arguments
+
+For function arguments, [annotate](https://peps.python.org/pep-0727/#specification) with the [`dagger.Doc`](https://dagger-io.readthedocs.io/en/latest/module.html#dagger.Doc) metadata.
+
+:::note
+[`dagger.Doc`](https://dagger-io.readthedocs.io/en/latest/module.html#dagger.Doc) is just an alias for [`typing_extensions.Doc`](https://typing-extensions.readthedocs.io/en/latest/#Doc).
+:::
+
+<!-- loud_hello has a multi-line docstring on purpose -->
+```python file=./snippets/documentation/python/main.py
+```
+
+</TabItem>
+<TabItem value="TypeScript">
+
+The following code snippet shows how to add documentation for:
+- The whole module
+- Function methods
+- Function arguments
+
+```typescript file=./snippets/documentation/typescript/index.ts
+```
+
+</TabItem>
+</Tabs>
+
+Here is an example of the result from `dagger functions`:
+
+```
+Name         Description
+hello        Return a greeting.
+loud-hello   Return a loud greeting.
+```
+
+Here is an example of the result from `dagger call hello --help`:
+
+```
+Return a greeting.
+
+Usage:
+  dagger call hello [flags]
+
+Flags:
+      --greeting string   The greeting to display (default "Hello")
+      --name string       Who to greet
+```
+
+
+The following code snippet shows how to add documentation for an object and its fields in your Dagger module:
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+```go file=./snippets/documentation/go/object.go
+```
+
+Here is an example of the result from `dagger call --help`:
+
+```
+Flags:
+      --age int
+      --name string
+      ...
+
+Function Commands:
+  age           The age of the user.
+  name          The name of the user.
+```
+
+</TabItem>
+<TabItem value="Python">
+
+```python file=./snippets/documentation/python/object.py
+```
+
+Here is an example of the result from `dagger call --help`:
```

Comment:
The Python output differs from that of the Go and TypeScript ones.
- For Go and TypeScript, the inline comments for object/struct properties appears in the output of dagger functions but not in the flag list of dagger call --help
- For Python, it's the opposite

Reply:
I'm guessing there is a fix needed to my code snippets, but I'm not sure what it is.

Reply:
Those are different things. `dagger functions` only show the available functions from selected object. It doesn't show arguments and that includes the constructor of course. It also doesn't show the description of the object that you're currently viewing, but `dagger call --help` should. So stick to `--help` as it shows all three:
- Main object's name and description and constructor arguments
- Currently selected function and description and available arguments
- Available functions to chain

Reply:
> I'm guessing there is a fix needed to my code snippets, but I'm not sure what it is.

I'll comment inline what I can notice just by the diff for now.